### PR TITLE
v3.5.0 - Fix CI bugs, dev refactoring, and issue fixes

### PR DESCRIPTION
**Pull request by DefinetlyNotAI**
### Prerequisites

- [x] I have [searched](https://github.com/DefinetlyNotAI/Logicytics/pulls) for duplicate or closed issues.
- [x] I have read the [contributing guidelines](https://github.com/DefinetlyNotAI/Logicytics/blob/main/CONTRIBUTING.md).
- [x] I have followed the instructions in the [wiki](https://github.com/DefinetlyNotAI/Logicytics/wiki) about contributions.
- [x] I have updated the documentation accordingly, if required.
- [x] I have tested my code with the `--dev` flag, if required.

### PR Type

- [x] Bug fix 
- [ ] Deprecation Change 
- [ ] New feature 
- [x] Refactoring 
- [ ] Documentation
  update 
- [ ] ⚠️ Breaking change ⚠️ 

### Description

This update changes the CI configuration to run jobs on Windows instead of Ubuntu. It includes running the `--dev` flag to update `config.ini` with new update info and testing Logicytics in default mode. The security documentation (`security.md`) has been updated, and a bug fix was applied to `dump_memory.py` to resolve a binary mode encoding error. Dev questions were refactored, and unimplemented flags were removed. File management and flag handling were refactored, along with an update to requirements for the `sentence-transformers` package. The README and installation instructions for the VulnScan feature have been updated. VulnScan tools and the v3 module were moved to a separate repository, retaining only the model and vectorizer in this repo. Lastly, issue labels and the CodeClimate badge were updated for better clarity.

### Motivation and Context

Switching CI jobs to Windows ensures better compatibility and aligns with target deployment environments. Refactoring and bug fixes improve code stability and maintainability. Moving VulnScan tools to a dedicated repo simplifies this project and clarifies separation of concerns. Updated documentation and requirements make setup easier for users and contributors, enhancing overall project quality and usability.

### Credit

_N/A_

### Issues Fixed

#228 

## Summary by CodeRabbit

* **Bug Fixes**
  * Corrected file handling in memory dump to ensure proper binary writing.
  * Updated prompts in developer tools for clearer user interaction.
  * Fixed return behavior in file management methods for consistency.
  * Adjusted error variable naming in packet sniffer for clarity.

* **New Features**
  * Updated security policy to add support for version 3.5.x.

* **Refactor**
  * Changed directory names for model and vectorizer files to lowercase for consistency.
  * Updated configuration and workflow files to reflect new naming conventions and environments.

* **Chores**
  * Removed deprecated or unused GitHub Actions workflows and configuration files.
  * Streamlined and updated dependencies in requirements.txt, removing unused packages and adjusting versions.
  * Removed planned but unimplemented command-line flags.

* **Documentation**
  * Updated README with improved PyTorch installation guidance and maintainability badge.
  * Removed obsolete VulnScan documentation and plans related to deprecated features.

### CODE REVIEW

File: CODE/dump_memory.py [hunk 1]
```diff
@@ -98,7 +98,7 @@ def memory_dump():
     try:
         process = psutil.Process(pid)
         dump_path = os.path.join(DUMP_DIR, "Ram_Dump.txt")
-        with open(dump_path, "wb", encoding="utf-8") as dump_file:
+        with open(dump_path, "wb") as dump_file:
```
[the remark this answers is not in the document]
Comment (DefinetlyNotAI) [reply CODE/dump_memory.py:101]: This has been fixed in the latest commits, after testing it out the bug was found and fixed

File: CODE/logicytics/FileManagement.py [hunk 2]
```diff
@@ -30,6 +30,7 @@ def open_file(file: str, use_full_path: bool = False) -> str | None:
                 subprocess.run(["start", file_path], shell=False)
             except Exception as e:
                 return f"Error opening file: {e}"
+        return None
```
[the remark this answers is not in the document]
Comment (DefinetlyNotAI) [reply CODE/logicytics/FileManagement.py:33]: A none return changes nothing much, but makes the code follow standards (PyCharm PEP)

File: CODE/vulnscan.py [hunk 3]
```diff
@@ -4,12 +4,12 @@
 import os
 import threading
 import warnings
-from pathlib import Path
 
 import aiofiles
 import joblib
 import numpy as np
 import torch
+from pathlib import Path
```
[the remark this answers is not in the document]
Comment (DefinetlyNotAI) [reply CODE/vulnscan.py:12]: Is not a major concern

File: CODE/vulnscan.py [hunk 4]
```diff
@@ -184,7 +184,7 @@ async def scan_worker(scan_file):
             "C:\\Program Files",
             "C:\\Program Files (x86)"
         ]
-        vulnscan = VulnScan("VulnScan/Model SenseMini .3n3.pth", "VulnScan/Vectorizer .3n3.pkl")
+        vulnscan = VulnScan("vulnscan/Model SenseMini .3n3.pth", "vulnscan/Vectorizer .3n3.pkl")
         vulnscan.scan_directory(base_paths)
```
[the remark this answers is not in the document]
Comment (DefinetlyNotAI) [reply CODE/vulnscan.py:188]: Path is correct, but will rename the files in the future

File: .github/workflows/scorecard.yml [hunk 5]
```diff
@@ -20,7 +20,7 @@ permissions: read-all
 jobs:
   analysis:
     name: Scorecard analysis
-    runs-on: ubuntu-latest
+    runs-on: windows-latest
```
[the remark this answers is not in the document]
Comment (DefinetlyNotAI) [reply .github/workflows/scorecard.yml]: Oof good catch
